modules/features2d/src/sift.dispatch.cpp

```diff
@@ -459,10 +472,14 @@ static void calcDescriptors(const std::vector<Mat>& gpyr, const std::vector<KeyP
 //////////////////////////////////////////////////////////////////////////////////////////
 
 SIFT_Impl::SIFT_Impl( int _nfeatures, int _nOctaveLayers,
-           double _contrastThreshold, double _edgeThreshold, double _sigma, int _descriptorType )
+           double _contrastThreshold, double _edgeThreshold, double _sigma, int _descriptorType, bool _enable_precise_upscale)
     : nfeatures(_nfeatures), nOctaveLayers(_nOctaveLayers),
-    contrastThreshold(_contrastThreshold), edgeThreshold(_edgeThreshold), sigma(_sigma), descriptor_type(_descriptorType)
+    contrastThreshold(_contrastThreshold), edgeThreshold(_edgeThreshold), sigma(_sigma), descriptor_type(_descriptorType),
+    enable_precise_upscale(_enable_precise_upscale)
 {
+    if (!enable_precise_upscale) {
+        CV_LOG_WARNING(NULL, "precise upscale disabled, this is now deprecated as it was found to induce a location bias");
```

Comment:
Users don't want messages "spam" even if they keep using legacy modes due to some reasons.
Especially if it is an "explicit" request from users as **this patch changes the default behavior** (and there is option to roll back only).
So please use `CV_LOG_ONCE_INFO()` here instead (ONCE with INFO level).